lib/StaticUniformGrid.js

```diff
@@ -0,0 +1,477 @@
+ * @alias StaticUniformGrid
+ * @constructor
+ *
+ * @param {Object[]} objects Objects that the grid should store.
+ * @param {Number} cellWidth, The cell width of the uniform grid.
+ * @param {StaticUniformGrid~CompareObjectBoundingBox} compareObjectBoundingBox
+ * @param {StaticUniformGrid~CellCheckFunction} cellCheck
+ */
+
+var minScratch = new Cartesian3();
+var maxScratch = new Cartesian3();
+var halfDimensionsScratch = new Cartesian3();
+
+function StaticUniformGrid(data, cellWidth, compareObjectBoundingBox, cellCheck) {
+    /**
+     * The width of a single cell.
+     * @type {Number}
+     */
+    this._cellWidth = cellWidth;
+
+    /**
+     * The inverse of the width of a single cell.
+     * @type {Number}
+     */
+    this._inverseCellWidth = 1.0 / cellWidth;
+
+    minScratch.x = Number.POSITIVE_INFINITY;
+    minScratch.y = Number.POSITIVE_INFINITY;
+    minScratch.z = Number.POSITIVE_INFINITY;
+    maxScratch.x = Number.NEGATIVE_INFINITY;
+    maxScratch.y = Number.NEGATIVE_INFINITY;
+    maxScratch.z = Number.NEGATIVE_INFINITY;
+
+    /**
+     * The axis aligned bounding box of the grid
+     * @type {AxisAlignedBoundingBox}
+     */
+    this._axisAlignedBoundingBox = new AxisAlignedBoundingBox(minScratch, maxScratch);
+
+    /**
+     * The cell count in each dimension.
+     * @type {Cartesian3}
+     */
+    this._resolution = new Cartesian3();
+
+    /**
+     * Start index for each cell's data within the data array.
+     * @type {Number[]}
+     */
+    this._cellIndices = [];
+
+    /**
+     * Count for each cell's data within the data array.
+     * @type {Number[]}
+     */
+    this._cellCounts = [];
+
+    /**
+     * Array of objects
+     * @type {Array}
+     */
+    this._data = [];
+
+    ////// Populate //////
+    var min = this._axisAlignedBoundingBox.minimum;
+    var max = this._axisAlignedBoundingBox.maximum;
+    var center = this._axisAlignedBoundingBox.center;
+
+    var resolution = this._resolution;
+
+    // Find the min/max bounds and resolution of the uniform grid
+    var objectCount = data.length;
+    for (var i = 0; i < objectCount; ++i) {
+        compareObjectBoundingBox(data[i], min, max);
+    }
+
+    var inverseCellWidth = this._inverseCellWidth;
+    // Figure out what the grid's resolution should be. Pad min and max out to match
+    resolution.x = Math.floor((max.x - min.x) * inverseCellWidth) + 1;
+    resolution.y = Math.floor((max.y - min.y) * inverseCellWidth) + 1;
+    resolution.z = Math.floor((max.z - min.z) * inverseCellWidth) + 1;
+
+    center = Cartesian3.add(min, max, center);
+    center = Cartesian3.divideByScalar(center, 2.0, center);
+
+    halfDimensionsScratch.x = (resolution.x * 0.5) * cellWidth;
+    halfDimensionsScratch.y = (resolution.y * 0.5) * cellWidth;
+    halfDimensionsScratch.z = (resolution.z * 0.5) * cellWidth;
+
+    min = Cartesian3.subtract(center, halfDimensionsScratch, min);
+    max = Cartesian3.add(center, halfDimensionsScratch, max);
```

Comment:
This was for updating the grid AABB's internal minimum and maximum, but it'll be more clear as
` this._axisAlignedBoundingBox.minimum = ...`